internal/host/refresh_status_preprocessor.go

```diff
@@ -30,35 +30,48 @@ type ValidationsStatus map[string][]ValidationResult
 type ValidationResults []ValidationResult
 
 type refreshPreprocessor struct {
-	log          logrus.FieldLogger
-	validations  []validation
-	conditions   []condition
-	operatorsApi operators.API
+	log                     logrus.FieldLogger
+	validations             []validation
+	conditions              []condition
+	operatorsApi            operators.API
+	disabledHostValidations DisabledHostValidations
 }
 
-func newRefreshPreprocessor(log logrus.FieldLogger, hwValidatorCfg *hardware.ValidatorCfg, hwValidator hardware.Validator, operatorsApi operators.API, disabledHostValidations []string) *refreshPreprocessor {
+func newRefreshPreprocessor(log logrus.FieldLogger, hwValidatorCfg *hardware.ValidatorCfg, hwValidator hardware.Validator, operatorsApi operators.API, disabledHostValidations DisabledHostValidations) *refreshPreprocessor {
 	v := &validator{
 		log:            log,
 		hwValidatorCfg: hwValidatorCfg,
 		hwValidator:    hwValidator,
 		operatorsAPI:   operatorsApi,
 	}
 	return &refreshPreprocessor{
-		log:          log,
-		validations:  newValidations(v, disabledHostValidations),
-		conditions:   newConditions(v),
-		operatorsApi: operatorsApi,
+		log:                     log,
+		validations:             newValidations(v),
+		conditions:              newConditions(v),
+		operatorsApi:            operatorsApi,
+		disabledHostValidations: disabledHostValidations,
 	}
 }
 
+const validationDisabledByConfiguration = "Validation disabled by configuration"
+
 func (r *refreshPreprocessor) preprocess(c *validationContext) (map[string]bool, ValidationsStatus, error) {
 	stateMachineInput := make(map[string]bool)
 	validationsOutput := make(ValidationsStatus)
 	for _, v := range r.validations {
-		st := v.condition(c)
-		stateMachineInput[v.id.String()] = st == ValidationSuccess
-		message := v.formatter(c, st)
 
+		var st ValidationStatus
+		var message string
+
+		if r.disabledHostValidations.IsDisabled(v.id) {
+			st = ValidationDisabled
+			message = validationDisabledByConfiguration
+			stateMachineInput[v.id.String()] = true
```

Comment:
Will do.

Reply:
Renamed to `conditions`, since it is the name of the receiving variable when this method is called.
https://github.com/openshift/assisted-service/blob/4780dc7e941c224a79d8ea8f1e6d5797cff2259b/internal/host/host.go#L324